### PR TITLE
Feature 305 quarkus 3 upgrade

### LINKED ISSUE #305: Qvarkus 3 upgrade
Framework update to Quarkus3

### PR DESCRIPTION
**Pull request by mrhaekkinen**
Resolves #305

### CODE REVIEW

File: src/main/kotlin/fi/metatavu/metaform/server/persistence/model/StringReplyField.kt [hunk 1]
```diff
@@ -15,7 +14,9 @@ import javax.persistence.Lob
 @Cacheable(true)
 class StringReplyField : ReplyField() {
   @Lob
+/*
```
Comment (anttileppa) [inline src/main/kotlin/fi/metatavu/metaform/server/persistence/model/StringReplyField.kt:17]: Removed commented code

File: src/main/kotlin/fi/metatavu/metaform/server/persistence/model/notifications/EmailNotification.kt [hunk 2]
```diff
@@ -22,14 +21,14 @@ class EmailNotification : fi.metatavu.metaform.server.persistence.model.Metadata
   var id: UUID? = null
 
   @Lob
-  @Type(type = "org.hibernate.type.TextType")
+  /*@Type(type = "org.hibernate.type.TextType")*/
```
Comment (anttileppa) [inline src/main/kotlin/fi/metatavu/metaform/server/persistence/model/notifications/EmailNotification.kt:24]: Remove commented code

File: src/main/kotlin/fi/metatavu/metaform/server/persistence/model/notifications/EmailNotification.kt [hunk 3]
```diff
@@ -22,14 +21,14 @@ class EmailNotification : fi.metatavu.metaform.server.persistence.model.Metadata
...
   @Column
   @NotNull
   @NotEmpty
   lateinit var subjectTemplate: String
 
   @Lob
-  @Type(type = "org.hibernate.type.TextType")
+  /*@Type(type = "org.hibernate.type.TextType")*/
```
Comment (anttileppa) [inline src/main/kotlin/fi/metatavu/metaform/server/persistence/model/notifications/EmailNotification.kt:31]: Remove commented code

File: src/test/java/fi/metatavu/metaform/server/test/functional/tests/MetaformStatisticsTestsIT.kt [hunk 4]
```diff
@@ -36,8 +37,19 @@ class MetaformStatisticsTestsIT: AbstractTest() {
...
+                    actualDateTime = metaformStatistics1.lastReplyDate!!,
+                    ChronoUnit.MILLIS
+            )
+            assertOffsetDateTimeEquals(
+                    expectedDateTime = reply2.createdAt!!,
+                    actualDateTime = metaformStatistics2.lastReplyDate!!,
+                    ChronoUnit.MILLIS
+            )
+
+            /*assertEquals(OffsetDateTime.parse(reply1.createdAt), OffsetDateTime.parse(metaformStatistics1.lastReplyDate))
```
Comment (anttileppa) [inline src/test/java/fi/metatavu/metaform/server/test/functional/tests/MetaformStatisticsTestsIT.kt:51]: Remove commented code

File: src/main/kotlin/fi/metatavu/metaform/server/persistence/model/Attachment.kt [hunk 5]
```diff
@@ -34,7 +33,7 @@ class Attachment : Metadata() {
 
   @Column(nullable = false)
   @Lob
-  @Type(type = "org.hibernate.type.BinaryType")
+  /*@Type(type = "org.hibernate.type.BinaryType")*/
```
[the remark this answers is not in the document]
Comment (mrhaekkinen) [reply src/main/kotlin/fi/metatavu/metaform/server/persistence/model/Attachment.kt:36]: Removed

File: src/main/kotlin/fi/metatavu/metaform/server/persistence/model/ExportTheme.kt [hunk 6]
```diff
@@ -30,7 +29,7 @@ class ExportTheme : Metadata() {
   var parent: ExportTheme? = null
 
   @Lob
-  @Type(type = "org.hibernate.type.TextType")
+  /*@Type(type = "org.hibernate.type.TextType")*/
```
[the remark this answers is not in the document]
Comment (mrhaekkinen) [reply src/main/kotlin/fi/metatavu/metaform/server/persistence/model/ExportTheme.kt:32]: Removed

File: src/main/kotlin/fi/metatavu/metaform/server/persistence/model/StringReplyField.kt (hunk 1 above)
Comment (mrhaekkinen) [reply src/main/kotlin/fi/metatavu/metaform/server/persistence/model/StringReplyField.kt:17]: Removed

File: src/main/kotlin/fi/metatavu/metaform/server/persistence/model/notifications/EmailNotification.kt (hunk 2 above)
Comment (mrhaekkinen) [reply src/main/kotlin/fi/metatavu/metaform/server/persistence/model/notifications/EmailNotification.kt:24]: Removed

File: src/main/kotlin/fi/metatavu/metaform/server/persistence/model/notifications/EmailNotification.kt (hunk 3 above)
Comment (mrhaekkinen) [reply src/main/kotlin/fi/metatavu/metaform/server/persistence/model/notifications/EmailNotification.kt:31]: Removed

File: src/main/kotlin/fi/metatavu/metaform/server/email/mailgun/MailgunEmailProviderImpl.kt [hunk 7]
```diff
@@ -43,37 +41,28 @@ class MailgunEmailProviderImpl : EmailProvider {
...
-                return
-            }
-        }
-        val response = mailBuilder.build().send()
-        if (response.isOk) {
-            logger.info("Send email to {}", toEmail)
-        } else {
-            logger.info("Sending email to {} failed with message {}", toEmail, response.responseMessage())
-        }
+        val client: WebClient = WebClient.create(vertx)
```
[the remark this answers is not in the document]
Comment (mrhaekkinen) [reply src/main/kotlin/fi/metatavu/metaform/server/email/mailgun/MailgunEmailProviderImpl.kt:55]: Removed formatting.

File: src/main/docker/Dockerfile.native [hunk 8]
```diff
@@ -14,8 +14,7 @@
 # docker run -i --rm -p 8080:8080 quarkus/metaform-api
 #
 ###
-FROM registry.access.redhat.com/ubi8/ubi-minimal:8.6
-RUN microdnf install freetype fontconfig
```
[the remark this answers is not in the document]
Comment (mrhaekkinen) [reply src/main/docker/Dockerfile.native:18]: Added

File: build.gradle.kts [hunk 9]
```diff
@@ -43,44 +36,46 @@ dependencies {
...
     testImplementation("org.awaitility:awaitility:3.1.2")
     testImplementation("org.apache.pdfbox:pdfbox:2.0.26")
     testImplementation("org.apache.pdfbox:pdfbox-tools:2.0.26")
     testImplementation("com.github.dasniko:testcontainers-keycloak:$testContainersKeycloakVersion")
-    testImplementation("com.github.tomakehurst:wiremock-jre8:$wiremockVersion")
+    testImplementation("org.wiremock:wiremock:$wiremockVersion")
+
 
-    kapt("org.hibernate:hibernate-jpamodelgen:5.5.7.Final")
+    // compileOnly("org.hibernate:hibernate-jpamodelgen:6.2.13.Final")
```
Comment (anttileppa) [inline build.gradle.kts:69]: Removed commeted code

**Review by anttileppa — APPROVED** [review]

**Pull request merged**